Update instructions to update TMAP8 and Conda (#109)
idaholab/TMAP8: doc/content/getting_started/installation.md

> capabilities, improvements, and bugfixes. Weekly updates are recommended as, at minimum, the MOOSE

Reviewer: Missed a `mamba activate` higher up (GitHub won't let me pinpoint. Line 59.

Author: Great catch!

submodule within TMAP8 is updated up to several times a week.

!alert note title=Always Update TMAP8 and the Conda packages together.

> There is a tight dependency between libraries being provided by Conda, and what libraries TMAP8 depends on. Therefore, when you update one you should always update the other.

Reviewer: ```suggestion
There is a tight dependency between the libraries and applications provided by Conda, and the submodules that TMAP8 depends on. Therefore, when you update one, you should always update the other.
```

To update your conda environment, perform the following commands (assuming that your MOOSE-based development environment is named `moose`):

> ``` bash
> conda activate base
> conda env remove -n moose
> conda create -n moose moose-dev=2024.03.05
> conda activate moose
> ```

Reviewer: ```suggestion
!package! code
conda activate base
conda env remove -n moose
conda create -n moose moose-dev=__MOOSE_DEV__
conda activate moose
!package-end!
```

This trick will keep the package version updated continuously alongside the automated submodule updates, using the MooseDocs package extension.

Author: I somehow couldn't get `!package! code` to work, but hopefully that does the trick! Thanks!

Reviewer: Let's see what CI says. I double-checked the default file it would reference (which is within the MOOSE repo, selected via MOOSE_DIR which defaults to the submodule), and also didn't see a requirement for explicit activation like some of the other extensions. I'll keep an eye on it.

> To update your TMAP8 repository as a TMAP8 user, use the following commands, which provide general users the most stable branch (`upstream/main`):

Reviewer: ```suggestion
To update your TMAP8 repository as a TMAP8 user, use the following commands, which provide to general users the content of the most stable branch (`upstream/main`):
```

```bash

> cd ~/projects/TMAP8

Reviewer: We don't mention doing a `git submodule update moose` post TMAP8 update, and probably should. Else, they won't have updated the MOOSE source code, necessarily.

capabilities, improvements, and bugfixes. Weekly updates are recommended as, at minimum, the MOOSE
submodule within TMAP8 is updated up to several times a week.

!alert note title=Always update TMAP8 and the Conda packages together.
There is a tight dependency between the libraries and applications provided by Conda, and the submodules that TMAP8 depends on. Therefore, when you update one, you should always update the other.

To update your conda environment, perform the following commands (assuming that your MOOSE-based development environment is named `moose`):

!package! code
conda activate base
conda env remove -n moose
conda create -n moose moose-dev=__MOOSE_DEV__
conda activate moose
!package-end!

To update your TMAP8 repository as a TMAP8 user, use the following commands, which provide to general users the content of the most stable branch (`upstream/main`):

```bash
cd ~/projects/TMAP8
git checkout main
git fetch upstream
git rebase upstream/main

> git submodule update --init moose

Reviewer: ```suggestion
git submodule update moose
```

Shouldn't need the init here if they are updating.

git checkout devel
git fetch upstream
git rebase upstream/devel

> git submodule update --init moose

Reviewer: ```suggestion
git submodule update moose
```

Same.
